sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -36,6 +36,21 @@ public final class StorageBlobClient: PipelineClient {
         case latest = "2019-02-02"
     }
 
+    /// The global maximum number of managed transfers that will be executed concurrently by all `StorageBlobClient`
+    /// instances. The default value is `maxConcurrentTransfersDefaultValue`. To allow this value to be determined
+    /// dynamically based on current system conditions, set it to `maxConcurrentTransfersDynamicValue`.
+    public static var maxConcurrentTransfers: Int {
+        get { return manager.maxConcurrency }
+        set { manager.maxConcurrency = newValue }
+    }
+
+    /// The default value of `maxConcurrentTransfers`.
+    public static let maxConcurrentTransfersDefaultValue = 4
+
+    /// Set `maxConcurrentTransfers` equal to this value to allow the maximum number of managed transfers to be
+    /// determined dynamically based on current system conditions.
+    public static let maxConcurrentTransfersDynamicValue = OperationQueue.defaultMaxConcurrentOperationCount
```

Comment:
That is an excellent question. So we're using `OperationQueue` internally, and in fact if you don't specify a max concurrent operation count for it, or if you explicitly set its value to `OperationQueue.defaultMaxConcurrentOperationCount`, that's what you get.

However, no one seems to really be sure what the dynamic value looks like. Is it in the range of 2-8 (depending on CPU load, hardware, etc), or more like 8-80? There is no documentation and the [one post](https://stackoverflow.com/questions/10441681/nsoperationqueuedefaultmaxconcurrentoperationcount-in-the-wild) I saw where someone was asking for observed behavior in the wild came up with no answers. However there are a few examples where people suggest that setting this to an explicit value provides significant performance benefit ([1](https://stackoverflow.com/questions/14995801/default-value-of-maxconcurrentoperationcount-for-nsoperationqueue), [2](https://github.com/SDWebImage/SDWebImage/pull/897)) at the expense of increased memory pressure.

FWIW, AlamoFire *does* use this as the default value: https://github.com/Alamofire/Alamofire/blob/master/Source/OperationQueue%2BAlamofire.swift#L38

Reply:
I tried using the default, and, at least with the CoreData instabilities, it fails very quickly. Once the CoreData issues are cleared up, I think it will make sense to let the system decide.